src/providers/types/content.rs

```diff
@@ -0,0 +1,127 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+// Base trait for all content types
```

Comment:
I'm not sure we need a ContentType here if we'll have them all be members of the enum below. It might be a little easier to understand if we just define the structs and then below something like

```rust
impl Content {
    pub fn summary(&self) -> String {
        match self {
            Content::Text(t) => format!("content:text\n{}", t.text),
            Content::ToolUse(t) => format!(
                "content:tool_use:{}\nparameters:{}",
                t.name,
                serde_json::to_string(&t.parameters).unwrap_or_default()
            ),
            Content::ToolResult(t) => format!(
                "content:tool_result:error={}\noutput:{}",
                t.is_error, t.output
            ),
        }
    }
```

This is a place i think rust is more elegant than the original python - since downstream looks like we're already handling most logic with `match` anyways. Also don't need the to_json methods because we can always do `serde_json::to_value(&content)` i believe

But i could be missing something if the polymorphism is used elsewhere!